### DISCUSSION TITLE
protect site_admin with fail2ban

### DISCUSSION THREAD

**Discussion by comsyspro** [LiveHelperChat/livehelperchat#1830] [Ideas]
i want to protect site_admin with fail2ban but i need a logfile where failed login attempts are logged. where does livehelperchat log the infos for failed login tries? or do you have a fail2ban rule which one can use?

**Comment by remdex** [comment]
Most clients just block `site_admin` for admin and that's all. See this table `lh_audits` this table stores failed logins. but only if valid username was passed.

**Reply by comsyspro** [reply]
>Most clients just block site_admin for admin and that's all.

do you mean to block it by e.g. .htacces basic authentication? or how to do it so that also operators (non admins) can login? thanks fo a hint.

**Reply by remdex** [reply]
Yes, by .htaccess. Usually, they allow just for a specific IP ranges.

So allow /site_admin/ just for the specific IP range. For operators and admins.

**Reply by comsyspro** [reply]
thank you. i also looked into the login.php file and there might be a possibility to append a logging command to put failed logins into a seperate log file, which can be watched by fail2ban.

**Reply by remdex** [reply]
Sure :) Create a pull request and if's standard PHP function I'll accept a pull request :)

**Comment by remdex** [comment]
You can also use 2FA methods to protect back office.

**Reply by comsyspro** [reply]
thanks, i tried it and 2fa also makes it secure. i also have found a solution now for fail2ban which protects the admin_site from failed login tries and bans the source ips and prevents attacks.
